### PR TITLE
BTFS-684: added a function to get metadata dag root

### PR DESCRIPTION
**Pull request by steveyeom**
1. Added  a new function 
2. Added a test to verify it.
3. All the unit tests passed without error.

### CODE REVIEW

File: test/utils.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ type NodeOpts struct {
 	rsNumParity        uint64
 	metadata           []byte
 	chunkSize          uint64
+	Balanced           bool
```
Comment (taiyangc) [inline test/utils.go]: Move to below `ReedSolomonEnabled` and add a short description

File: unixfs.go [hunk 2]
```diff
@@ -412,3 +416,79 @@ func ExtractFSNode(node ipld.Node) (*FSNode, error) {
...
+
+// Return ipld.Node slice of size 2
+// if the given 'nd' is top of the DAG with token metadata.
+// Return nil, nil if 'nb' is no such node and there is no error.
+func GetChildrenForDagWithMeta(nd ipld.Node, ds ipld.DAGService) ([]ipld.Node, error) {
+	var nodes = make([]ipld.Node, 2)
+	for i := 0; i < 2; i++ {
+		lnk := nd.Links()[i]
+		c := lnk.Cid
+		child, err := ds.Get(context.Background(), c)
```
Comment (taiyangc) [inline unixfs.go:462]: Pass `ctx context.Context` as a function argument. So upper level can control timeouts/cancellations

File: unixfs.go [hunk 3]
```diff
@@ -412,3 +416,79 @@ func ExtractFSNode(node ipld.Node) (*FSNode, error) {
...
+func GetMetaSubdagRoot(n ipld.Node, serv ipld.NodeGetter) (ipld.Node, error) {
+	nd, ok := n.(*dag.ProtoNode)
+	if !ok {
+		return nil, dag.ErrNotProtobuf
+	}
+
+	if nd.Links() != nil && len(nd.Links()) >= 2 {
+		link := nd.Links()[0]
+		c := link.Cid
+		child, err := serv.Get(context.Background(), c)
```
Comment (taiyangc) [inline unixfs.go:431]: Pass context from function argument

File: unixfs.go (hunk 2 above)
Comment (steveyeom) [reply unixfs.go:462]: Done

File: unixfs.go (hunk 3 above)
Comment (steveyeom) [reply unixfs.go:431]: Done

**Comment by steveyeom** [comment]
Performed go-btfs unit tests and go-unixfs unit tests. All passed.

**Review by taiyangc — APPROVED** [review]
looks good

**Pull request merged**